src/Cli/dotnet/commands/dotnet-workload/install/NetSdkMsiInstallerClient.InstallRecords.cs

```diff
@@ -0,0 +1,342 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Text.Json;
+using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Installer.Windows;
+using Microsoft.NET.Sdk.WorkloadManifestReader;
+using Microsoft.Win32;
+using Microsoft.Win32.Msi;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+using static Microsoft.NET.Sdk.WorkloadManifestReader.WorkloadResolver;
+
+namespace Microsoft.DotNet.Workloads.Workload.Install
+{
+    internal partial class NetSdkMsiInstallerClient
+    {
+        //  Manifest IDs are lowercased on disk, we need to map them back to the original casing to generate the right UpgradeCode
+        private static readonly string[] CasedManifestIds =
+            [
+                "Microsoft.NET.Sdk.Android",
+                "Microsoft.NET.Sdk.Aspire",
+                "Microsoft.NET.Sdk.iOS",
+                "Microsoft.NET.Sdk.MacCatalyst",
+                "Microsoft.NET.Sdk.macOS",
+                "Microsoft.NET.Sdk.Maui",
+                "Microsoft.NET.Sdk.tvOS",
+                "Microsoft.NET.Workload.Emscripten.Current",
+                "Microsoft.NET.Workload.Emscripten.net6",
+                "Microsoft.NET.Workload.Emscripten.net7",
+                "Microsoft.NET.Workload.Mono.ToolChain.Current",
+                "Microsoft.NET.Workload.Mono.ToolChain.net6",
+                "Microsoft.NET.Workload.Mono.ToolChain.net7",
```

Comment:
@dsplaisted I assume this list will need to be updated to add the `.net8` packages in main right?

Reply:
@akoeplinger This list of IDs is actually part of a workaround for the fact that the manifest MSIs weren't writing a manifest install record to the registry.  We added that in https://github.com/dotnet/arcade/pull/14359, so the .net8 manifest MSIs (which are produced in main) should have the right installation records and shouldn't need this fallback logic to be detected.

Reply:
Perfect, thanks